### PR TITLE
Add exit signal handler to casync-http

### PR DESCRIPTION
**Pull request by elboulangero**
When hitting "Ctrl-C" during `casync extract`, the casync-http worker receives a signal which is not handled, resulting in an unclean exit: the `/var/tmp/<...>.carem` directory is not deleted.

This commit adds basic signal handling for exit signals, ensuring clean exit and deletion of the temporary work directory in `/var/tmp`.

### CODE REVIEW

File: src/casync-http.c [hunk 1]
```diff
@@ -453,6 +455,11 @@ static int run(int argc, char *argv[]) {
                 const char *store_url;
                 CaChunkID id;
 
+                if (quit) {
+                        log_info("Got exit signal, quitting.");
```
Comment (mihais) [inline src/casync-http.c]: Please insert an 'r = 0;' before the goto just to set the exit value as SUCCESS.

Comment (elboulangero) [reply src/casync-http.c]: Done, thanks for pointing that out.

File: src/casync-http.c [hunk 2]
```diff
@@ -10,6 +10,8 @@
 #include "realloc-buffer.h"
 #include "util.h"
 
+volatile sig_atomic_t quit = false;
```
Comment (poettering) [inline src/casync-http.c]: static please

Comment (elboulangero) [reply src/casync-http.c]: of course. pushed.

**Pull request merged**